### PR TITLE
Handle fleet manager REST api errors

### PR DESCRIPTION
**Pull request by SimonBaeumer**
## Description
Handle fleet manager API errors correctly. 

## Checklist (Definition of Done)

- [ ] ~~Unit and integration tests added~~
- [ ] ~~Documentation added if necessary~~
- [ ] CI and all relevant tests are passing

## Test manual

 - Run Fleet-Manager + Fleetshard-sync with an invalid OCM token, should write errors into the log

### CODE REVIEW

File: fleetshard/pkg/runtime/runtime.go [hunk 1]
```diff
@@ -88,12 +89,11 @@ func (r *Runtime) handleReconcileResult(central private.ManagedCentral, status *
 		return
 	}
 
-	resp, err := r.client.UpdateStatus(map[string]private.DataPlaneCentralStatus{
+	err = r.client.UpdateStatus(map[string]private.DataPlaneCentralStatus{
 		central.Id: *status,
 	})
 	if err != nil {
-		glog.Errorf("error occurred %s: %s", central.Metadata.Name, err.Error())
+		err = errors.Wrapf(err, "updating central status %s", central.Metadata.Name)
```
Comment (mtesseract) [inline fleetshard/pkg/runtime/runtime.go:96]:
```suggestion
		err = errors.Wrapf(err, "updating status for Central %q", central.Metadata.Name)
```

**Review by mtesseract — APPROVED** [review]
Mostly remarks on error messages.

**Review by kovayur — COMMENTED** [review]
This already looks good, I'll keep my approval until Moritz's feedback is addressed

**Review by johannes94 — APPROVED** [review]
Looks good.

File: fleetshard/pkg/fleetmanager/client.go [hunk 2]
```diff
@@ -57,39 +58,39 @@ func (c *Client) GetManagedCentralList() (*private.ManagedCentralList, error) {
...
-	if err != nil {
-		return nil, err
-	}
-
 	list := &private.ManagedCentralList{}
-	err = json.Unmarshal(respBody, &list)
+	err = c.unmarshalResponse(resp.Body, &list)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrapf(err, "error calling %s", c.endpoint)
```
[the remark this answers is not in the document]
Comment (SimonBaeumer) [reply fleetshard/pkg/fleetmanager/client.go:64]: done

File: fleetshard/pkg/fleetmanager/client.go [hunk 3]
```diff
@@ -57,39 +58,39 @@ func (c *Client) GetManagedCentralList() (*private.ManagedCentralList, error) {
...
 	resp, err := http.NewRequest(http.MethodPut, fmt.Sprintf("%s/%s", c.endpoint, statusRoute), bufUpdateBody)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	return ioutil.ReadAll(resp.Body)
+	into := make(map[string]interface{})
+	if err := c.unmarshalResponse(resp.Body, &into); err != nil {
+		return errors.Wrapf(err, "failed to updates status")
```
[the remark this answers is not in the document]
Comment (SimonBaeumer) [reply fleetshard/pkg/fleetmanager/client.go:91]: done

File: fleetshard/pkg/fleetmanager/client.go [hunk 4]
```diff
@@ -106,3 +107,28 @@ func (c *Client) newRequest(method string, url string, body io.Reader) (*http.Re
...
 	return resp, nil
 }
+
+func (c *Client) unmarshalResponse(body io.Reader, v interface{}) error {
+	data, err := ioutil.ReadAll(body)
+	if err != nil {
+		return err
+	}
+
+	into := make(map[string]interface{})
```
[the remark this answers is not in the document]
Comment (SimonBaeumer) [reply fleetshard/pkg/fleetmanager/client.go]: That's a good idea, added it

File: fleetshard/pkg/runtime/runtime.go [hunk 5]
```diff
@@ -59,7 +59,8 @@ func (r *Runtime) Start() error {
 	ticker := concurrency.NewRetryTicker(func(ctx context.Context) (timeToNextTick time.Duration, err error) {
 		list, err := r.client.GetManagedCentralList()
 		if err != nil {
-			glog.Error("failed to list central", err)
+			err = errors.Wrapf(err, "failed to list centrals")
```
[the remark this answers is not in the document]
Comment (SimonBaeumer) [reply fleetshard/pkg/runtime/runtime.go:66]: done

**Review by mtesseract — APPROVED** [review]

**Pull request merged**